### PR TITLE
[firejail] add and install our own whitelist

### PR DESCRIPTION
**Pull request by nephros**
See Issue #55 for discussion.

Installs a new file `whitelist-common-patchmanager.local`, and adds an include line to `/etc/firejail/whitelist-common.local`.

Tested on 4.0/Xperia10-DS and 4.2/GeminiPDA, works.

### CODE REVIEW

File: rpm/patchmanager.spec [hunk 1]
```diff
@@ -61,6 +61,8 @@ ln -s ../lipstick-patchmanager.service %{buildroot}/%{_userunitdir}/lipstick.ser
 
 mkdir -p %{buildroot}%{_datadir}/%{name}/patches
 
+install -m 644 -D src/etc/whitelist-common-patchmanager.local %{buildroot}%{_sysconfdir}/firejail/whitelist-common-patchmanager.local
```
Comment (nephros) [inline rpm/patchmanager.spec:64]: I use plain install here as I am not familiar how a .pro file would do this.

For sake of consistency it would probably be good to do this in a .pro file like everything else does.

**Review by Olf0 — COMMENTED** [review]
Looking good; but I have no practical experience with `pro` files (just looked at a few).

File: rpm/patchmanager.spec [hunk 2]
```diff
@@ -124,6 +129,7 @@ export NO_PM_PRELOAD=1
 case "$*" in
 0)
 echo "Uninstalling %{name}: postun section"
+sed -i "/whitelist-common-patchmanager.local/ d" /etc/firejail/whitelist-common.local ||:
```
[the remark this answers is not in the document]
Comment (Olf0) [reply rpm/patchmanager.spec:132]: Yes, please: This is a lot easier to understand, when reading it the first time.
And `|| true` (with a space after the `||`) is now consistent with all other occurrances of this pattern in the spec file.

[the remark this answers is not in the document]
Comment (Olf0) [reply rpm/patchmanager.spec]: I just wondered, because the other `sed -i "/xyz/ d" foo`-lines did not have a trailing `|| true`!
And remembered that `rm -f` is (by definition) always successful, even if the path is invalid.
Tested => sed -i and rm -f always return "success"!

This leaves a single trailing `|| true` in [line 77](https://github.com/sailfishos-patches/patchmanager/blob/master/rpm/patchmanager.spec#L77), as I expect Patchmanager to be able to fail with the `--unapply-all` option (if it cannot, we should fix that in the long run; at least this clearly denotes this expectation).

[the remark this answers is not in the document]
Comment (Olf0) [reply rpm/patchmanager.spec]: Plus a some other, little nitpicks in `Keep quoting style and use of %{name} consistent` and `Omit output, when all is fine`.

**Comment by nephros** [comment]
Thanks for the thorough reviews!

Unfortunately I force-pushed before pulling, sorry about that.
I manually applied the changes as fixups because I don't know how to pull single commits.
Hope that went ok, please review once more.

File: rpm/patchmanager.spec (hunk 2 above)
[the remark this answers is not in the document]
Comment (nephros) [reply rpm/patchmanager.spec]: Hi, I'm absolutely fine with pushing to PRs, but my pre-coffee self force-pushed before integrating them so they got lost. Sorry.
Applied them manually again as plain diffs and fixup commits.

On the topic of `||:` vs `|| true`: `||:` is what the rest of the world (i.e. Fedora/SuSE) seems to use, that's where I picked it up from. But I have no preference either way.

**Review by b100dian — APPROVED** [review]

**Comment by Olf0** [comment]
> Hope that went ok, please review once more.

Looks fine.

File: rpm/patchmanager.spec (hunk 2 above)
[the remark this answers is not in the document]
Comment (Olf0) [reply rpm/patchmanager.spec]: > On the topic of `||:` vs `|| true`: `||:` is what the rest of the world (i.e. Fedora/SuSE) seems to use, that's where I picked it up from.

Thanks for denoting.
The point is that Fedora (and RHEL / CentOS) and (Open)SuSE use Bash as default shell, and I believe the built-in `:` is not available in all shells (would have to research that, to be sure).

When I stated using Bash as my primary interactive shell on GNU/Linux more than 25 years ago, I have been burnt by (accidentally) using Bashisms at BSDs, HP-UX, AIX, Solaris, IRIX etc. so many times, that I started to avoid them; strictly so when writing shell scripts.
A few years ago, I started to think "I am solely using Linux nowadays, and only very few distributions do not install Bash by default (e.g., OpenWRT on my routers), so I may become accustomed to Bashisms again", only to see SailfishOS switch to busybox's ash a few years later (due to the GPLv3 of recent Bash releases).
While busybox's ash supports some Bashisms (e.g., `set -o pipefail`), many are unsupported.

TL;DR: Do avoid Bashisms, especially in shell scripts, in order to make them portable.
Here: `|| true` (or `||true`) is guaranteed to work everywhere (and `true` is usually also a shell built-in command, hence using it does not make a difference), while `|| :` (or `||:`) needs checking for the specific shell in use.
